frontend/packages/dev-console/src/components/import/advanced/ProgressiveList.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import FormSectionDivider from '../section/FormSectionDivider';
+import './ProgressiveList.scss';
+
+export const getGeneratedText = (text, items: string[], callback) => {
+  if (!items || items.length === 0) {
+    return null;
+  }
+  return (
+    <p>
+      {text}
+      {items.map((opt, index) => {
+        let preText = ' ';
+        let postText = '';
+        if (items.length - 1 === index) {
+          preText = items.length !== 1 ? ' and ' : ' ';
+          postText = '.';
+        } else {
+          postText = items.length - 2 !== index ? ',' : '';
+        }
+        return (
+          <React.Fragment key={opt}>
+            {preText}
+            <button
+              className="btn btn-link odc-progressive-list__button"
+              type="button"
+              onClick={(event: React.MouseEvent<HTMLButtonElement>) => {
+                const target = event.target as HTMLButtonElement;
+                callback(target.innerText);
+              }}
+            >
+              {opt}
+            </button>
+            {postText}
+          </React.Fragment>
+        );
+      })}
+    </p>
+  );
+};
+
+type ProgressiveListProps = {
```

Comment:
Just for consistency since we are using interface everywhere else.
```suggestion
export interface ProgressiveListProps = {
```